xdsl/irdl/declarative_assembly_format.py

```diff
@@ -24,20 +26,24 @@ class ParsingState:
     It contains the elements that have already been parsed.
     """
 
+    operands: list[None | UnresolvedOperand]
```

Comment:
super merga ultra nit:
```suggestion
    operands: list[UnresolvedOperand | None]
```

This is so tiny I feel silly suggesting this change but None's always been the last type so far in unions

Reply:
No no, you're right, my eyes cried looking at this again.